#
# Submitted-By: Robert Schwebel, 2006-05-25
# Committed-By: Robert Schwebel
#
# Error:
#
# In member function 'ushort& QChar::unicode()':
# /build/buildd/qt4-x11-4.0.0/include/QtCore/../../src/corelib/tools/qchar.h:225:
# error: cannot bind packed field '((QChar*)this)->QChar::ucs' to 'ushort&' 
#
# Description:
#
# http://lists.debian.org/debian-arm/2005/12/msg00009.html
#
# State:
# 
# unknown
#

diff -urN qtopia-core-opensource-src-4.1.1/src/corelib/tools/qchar.h qtopia-core-opensource-src-4.1.1-ptx/src/corelib/tools/qchar.h
--- qtopia-core-opensource-src-4.1.1/src/corelib/tools/qchar.h	2006-02-15 14:03:33.000000000 +0100
+++ qtopia-core-opensource-src-4.1.1-ptx/src/corelib/tools/qchar.h	2006-04-18 15:26:33.000000000 +0200
@@ -219,7 +219,7 @@
     inline const char toLatin1() const;
     inline const ushort unicode() const { return ucs; }
 #ifdef Q_NO_PACKED_REFERENCE
-    inline ushort &unicode() { return *(&ucs); }
+    inline ushort &unicode() { return *((ushort*)&ucs); }
 #else
     inline ushort &unicode() { return ucs; }
 #endif
